### PR TITLE
Add support for the OpenTitan chip

### PR DESCRIPTION
**Pull request by nasahlpa**
This PR adds support for flashing, resetting, and communicating with the discrete OpenTitan chip.

### CODE REVIEW

File: target/chip.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                               "--exec", "transport init",
+                               "--exec", "bootstrap " + self.firmware, "no-op"],
+                              stdout=PIPE, stderr=PIPE)
+        flash_process.communicate()
+        rc = flash_process.returncode
+        if rc != 0:
+            raise RuntimeError('Error: Failed to flash OpenTitan chip.')
+        else:
+            # Wait until chip finished booting.
+            time.sleep(1)
```
Comment (vrozic) [inline target/chip.py:32]: How sure are we that 1s is enough to finish booting?

Comment (nasahlpa) [reply target/chip.py:32]: Good point - I've made some experiments and I think that 1s should be more than enough. For FI experiments, we might even reduce it to 500 or 750ms. I am changing the code such that the user can specify the boot delay.

**Review by vogelpi — APPROVED** [review]
This looks good, thanks @nasahlpa .

I think it would make sense to mention the exact commit you used to build the opentitantool binary. Writing something like https://github.com/lowRISC/OpenTitan/commit/ca93a12 should work (meaning you don't need to specify the branch name and full hash, 7 chars is fine) the important thing is that the commit is really upstreamed.

**Comment by nasahlpa** [comment]
Thanks for your comments, I've included the commit hash and command used to build the opentitantool.

**Pull request merged**